### PR TITLE
Add support for audio over USB

### PR DESCRIPTION
**Pull request by v3rm0n**
- Add possibility to choose an audio output device
- Use async libusb API so that it works better with audio

### CODE REVIEW

File: ringbuffer.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#ifndef M8C_RINGBUFFER_H
+#define M8C_RINGBUFFER_H
+
+#include <stdlib.h>
+
```
Comment (laamaa) [inline ringbuffer.h:5]: I unfortunately couldn't compile the code with cmake configuration, I think stdint.h should be included in this file and the respective .c file too to get uint32_t etc.?

**Comment by laamaa** [comment]
Thanks! :)
Could you please update the regular Makefile too to pick up the new files and add stdint.h to the ringbuffer's includes?

**Comment by laamaa** [comment]
(It might be a good idea to mention the possibility of specifying the device name in README.md too)

**Comment by v3rm0n** [comment]
Does it work now?

**Comment by laamaa** [comment]
Yep, thank you! :)

**Pull request merged**